src/Workspaces/Core/Portable/Shared/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Extensions
+{
+    internal static class CommonParenthesizedExpressionSyntaxExtensions
+    {
+        public static bool IsSafeToChangeAssociativity(
```

Comment:
:memo: (for other reviewers) This method is extracted from code in `RemovalChangesAssociation`, but has been updated to invert the meaning of the return value and handle some additional cases noted in the diff.